### PR TITLE
docs: Add RedwoodJS Quick Start

### PR DESCRIPTION
**Pull request by dthyresson**
This PR is the first draft of a Defer quick start with RedwoodJS.

Note that since I cannot run the Nextra docs, I haven't checked the layout etc.

I expect some tweaks to the docs once I have other RedwoodJS Core team members test them out and give feedback.

This PR also includes a few small changes to the `syncContacts` examples for other integrations.

### CODE REVIEW

File: pages/quickstart/redwoodjs.mdx [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+  +-- graphql
+  +== jobs                     // Create a new Jobs folder
+    +== clients
+      += defer.ts              // Configure and export the defer client
+    +== defer                  // Write your defer background, cron, delayed, and workflow functions here
+        +- <>
+  +-- lib
+      +- logger.ts
+  +-- services
+
```
Comment (charlypoly) [inline pages/quickstart/redwoodjs.mdx]:
```suggestion
.
├── api
│   ├── db/
│   ├── package.json
│   ├── server.config.js
│   ├── src/
│   │   ├── directives/
│   │   ├── functions/
│   │   │   ├── runHelloWorld/
│   │   │   │   └── runHelloWorld.ts <- API endpoint to invoke job
│   │   ├── graphql/
│   │   ├── jobs/ <- Create a new Jobs folder
│   │   │   ├── clients
│   │   │   │   └── defer.ts <- Configure and export the defer client
│   │   │   └── defer/ <- Write your defer background, cron, delayed, and workflow functions here
│   │   │       └── helloWorld.ts
│   │   ├── lib/
│   │   │   └── logger.ts
│   │   └── services/
├── package.json
├── redwood.toml
├── web/
```

File: pages/quickstart/redwoodjs.mdx [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+<p>&nbsp;</p>
+
+### Create a `defer/` folder
+
+The Defer Platform expects to find all the background functions in the `defer/` folder.
+
+In a RedwoodJS app, we recommend creating a `jobs/` folder in your project's existing `api` side structure
+to organize both your defer functions and client as follows:
+
+```txt {4}
```
Comment (charlypoly) [inline pages/quickstart/redwoodjs.mdx]:
```suggestion
```txt {10, 12, 14, 15}
```

File: pages/quickstart/redwoodjs.mdx [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+
+<br />
+
+To test in local development, first start your RedwoodJS development server:
+
+```bash copy
+  yarn rw dev
+```
+
+Then, open your browser and visit the `http://localhost:8910/.redwood/functions/hellow?name=Charly` function endpoint.
```
Comment (charlypoly) [inline pages/quickstart/redwoodjs.mdx]:
```suggestion
Then, open your browser and visit the `http://localhost:8910/.redwood/functions/helloWorld?name=Charly` function endpoint.
```

Is it a typo or am I missing something? 👀

File: pages/quickstart/redwoodjs.mdx [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+
+If you are wondering how calling a background function triggers an execution on Defer Platform, please refer to this guide: [How Defer works](/platform/how-defer-works).
```
Comment (charlypoly) [inline pages/quickstart/redwoodjs.mdx]:
```suggestion
```

Move in the dedicated lonely `<Callout>` below

File: pages/quickstart/redwoodjs.mdx [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+
+and the response:
+
+```json
+{ "ok": true }
+```
+
+<Callout type={"info"}>
+**How does Defer work?**
+
```
Comment (charlypoly) [inline pages/quickstart/redwoodjs.mdx:243]:
```suggestion

If you are wondering how calling a background function triggers an execution on Defer Platform, please refer to this guide: [How Defer works](/platform/how-defer-works).
```

File: pages/quickstart/redwoodjs.mdx [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+      alt="Your first successful build on the Defer Console"
+      width={400}
+      height={450}
+    />
+  </div>
+</div>
+
+<Callout type={'info'}>
+**A Defer application needs a successful build to accept executions
+**
```
Comment (charlypoly) [inline pages/quickstart/redwoodjs.mdx]:
```suggestion
**A Defer application needs a successful build to accept executions**

```

File: pages/quickstart/redwoodjs.mdx [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
+    <Image
+      src="/getting-started/quickstart-redwoodjs-build-settings.png"
+      alt="Configure your Build Settings on the Defer Console"
+      width={800}
+      height={900}
+    />
+  </div>
+</div>
+
+### Configure your API deployment
```
Comment (charlypoly) [inline pages/quickstart/redwoodjs.mdx:313]:
```suggestion
<p>&nbsp;</p>
<p>&nbsp;</p>
<p>&nbsp;</p>

### Configure your API deployment
```

**Review by charlypoly — CHANGES_REQUESTED** [review]
@dthyresson did some minor UI and typos fixes  🦅 

Thank you very much for putting this guide down! 🙇🏼

File: pages/quickstart/redwoodjs.mdx (hunk 1 above)
Comment (charlypoly) [reply pages/quickstart/redwoodjs.mdx]: I updated it to be aligned with other guides

File: pages/quickstart/redwoodjs.mdx (hunk 3 above)
Comment (dthyresson) [reply pages/quickstart/redwoodjs.mdx]: Typo, bad copy pasting on my side. Thanks!

**Comment by dthyresson** [comment]
> @dthyresson did some minor UI and typos fixes 🦅
> 
> Thank you very much for putting this guide down! 🙇🏼

Thanks for the typo fixes and formatting ... it's tough to write mdx blindly without the actual render :)

I believe I committed all the suggestions.

**Review by charlypoly — APPROVED** [review]

**Comment by charlypoly** [comment]
@dthyresson are we still waiting for reviews from the RedwoodJS Core team? 👀

**Comment by dthyresson** [comment]
> @dthyresson are we still waiting for reviews from the RedwoodJS Core team? 👀

Yes. Thanks for reminding me. I’ve asked someone for a review today/tomorrow

**Comment by charlypoly** [comment]
@dthyresson, do you already know which command will be defined to generate a defer background function? (`yarn rw experimental defer function`) 

Asking for the upcoming Partner pitch 🚀

**Comment by dthyresson** [comment]
> @dthyresson, do you already know which command will be defined to generate a defer background function? (`yarn rw experimental defer function`)
> 
> Asking for the upcoming Partner pitch 🚀

Finished Core team review and good to publish. 👍 

At moment there will not be a dedicated defer command to generate a type of function as I haven't had a chance to work on it.

We plan on moving much out of experimental (like the inngest ones) and at that time can add defer function setup and generators, if that's ok?

It would be similar to: https://github.com/inngest/envelop-plugin-inngest/tree/main/packages/setup/redwoodjs for the functions -- but a little different for the sdk setups.

For the partner pitch you can mention in future RW will have setup and generator commands.

**Pull request merged**